cluster/src/main/scala/org/apache/spark/memory/SnappyUnifiedMemoryManager.scala

```diff
@@ -136,6 +136,37 @@ class SnappyUnifiedMemoryManager private[memory](
     } else memoryMap
   }
 
+
+  /**
+    * This method will be called if executor is going to be restarted.
+    * When executor is coming up all accounting from store will be done in
+    * bootMemoryManager.
+    * When executor stops we will copy the existing entry in this manager to
+    * boot manager.
+    * Once executor comes back again we will again copy the boot manager entries
+    * to run time manager.
+    */
+  override def close(): Unit = {
+    assert(!tempManager)
+    // First reset the memory manager in callback. Hence all request will
+    // go to Boot Manager
+    MemoryManagerCallback.resetMemoryManager()
+    synchronized {
+      logInfo(s" Closing Memory Manager ${this}")
+      val bootManager = MemoryManagerCallback.bootMemoryManager
+          .asInstanceOf[SnappyUnifiedMemoryManager]
+
+      memoryForObject.object2LongEntrySet().iterator().asScala foreach { entry =>
+        val (objectName, memoryMode) = entry.getKey
+        if (!objectName.equals(SPARK_CACHE) &&
+            !objectName.equals(BufferAllocator.STORE_DATA_FRAME_OUTPUT)) {
```

Comment:
endsWith for latter since off-heap uses DIRECT_STORE_DATA_FRAME_OUTPUT

Reply:
Will change it.